### ISSUE TITLE
Transaction fee 

### ISSUE THREAD

**Issue by kumrzz** [richardkiss/pycoin#198]
TX_FEE_PER_THOUSAND_BYTES = 10000

**Comment by dabura667** [comment]
Yes... That and also, the estimation based on tx size rounded up to the nearest 1000 bytes is old hat as well.

All miners have demonstrated they calculate satoshi per byte fees (no rounding) and mostly order by that.

**Comment by shayanb** [comment]
what do you think of this implementation for dynamic fees? 
https://github.com/pycoin/pycoin/blob/tx_fee/pycoin/convention/tx_fee.py

**Comment by richardkiss** [comment]
There are some good ideas here, and I think adding a method to the existing blockcypher service is in order. However, I don't like the idea of going to the network in a function inside the "convention" module. I think perhaps accepting a parameter "tx_fee_per_byte" is a better way to go. That way users could call the service to get the current expected fee, then call this revised function to get the exact fee. Anyone want to take a crack at it?

Getting the exact size for a transaction in general is, of course, very difficult. You have to estimate the size of the signed transaction, which can be tricky in the case of coins coming from, for example, a pay to script hash.

One way to do it would be to create the transaction with a placeholder fee, sign the transaction, count the exact number of bytes, then update the fee.

**Comment by shayanb** [comment]
@richardkiss I Agree. I think I can get it done tomorrow.

**Comment by dabura667** [comment]
> sign the transaction

This could be dangerous, I would recommend just assuming the signature length in the push. (maybe make a method called fake_sign() that could just insert dummy data that is 72 byte pushdata per sig.)

Since an ECDSA sig in the strict DER format with low S enforced including sighash byte can only be 72 bytes or 71 bytes pushdata. Assume highest and go with 72.

**Comment by dabura667** [comment]
Not to mention it is inconvenient to require private key handling to estimate fee properly.

**Comment by shayanb** [comment]
I spend a lot of time on this and couldn't figure out a good way to do it. I tried adding a dummy_sign function to add 72 bytes for each input, but even though I was doing this on another copy of TX object, the final signed TX was invalid (invalid inputs). 

I think I'd just implement something on the application layer and not on pycoin for now.
